src/kudu/consensus/raft_consensus.cc

```diff
@@ -4009,6 +4012,11 @@ void RaftConsensus::SetLeaderDetectedCallback(LeaderDetectedCallback ldcb) {
   ldcb_ = std::move(ldcb);
 }
 
+void RaftConsensus::SetVoteLogger(std::shared_ptr<VoteLoggerInterface> vote_logger) {
+  CHECK(vote_logger);
```

Comment:
Should we be checking this? Do we intend to allow voter logger to be cleared?

Reply:
I am thinking in the future we can make a sys_var to disable logger, so null logger should be allowed?

Reply:
If null logger is allowed CHECK(vote_logger) will immediately crash here right?

Reply:
Yeah. I think CHECK() is only used in debug build. I think production build / optimal build should ignore CHECK. (at least for mysqld)

Reply:
CHECK() will crash in prod/opt too. If you only want it in debug builds, use DCHECK()

Reply:
Oh I see. My bad. I thought you suggested using "CHECK()". It is actually I have "CHECK()" and you suggested removing it.

We should not use CHECK() to allow it to be turn off. Removed.